.github/workflows/build-tarballs.yml

```diff
@@ -0,0 +1,56 @@
+name: Build tarballs for testing
+on:
+  pull_request:
+    branches: [master]
+
+jobs:
+  build-tarball:
+    runs-on: 'ubuntu-latest'
+    strategy:
+      matrix:
+        package: ['builder', 'eslint-plugin-foreman', 'find-foreman', 'stories', 'test', 'vendor-core', 'vendor-dev', 'vendor']
+        include: # providing package path info
+        - package: builder
+          path: 'packages/builder/**'
+        - package: eslint-plugin-foreman
+          path: 'packages/eslint-plugin-foreman/**'
+        - package: find-foreman
+          path: 'packages/find-foreman/**'
+        - package: stories
+          path: 'packages/stories/**'
+        - package: test
+          path: 'packages/test/**'
+        - package: vendor-core
+          path: 'packages/vendor-core/**'
+        - package: vendor-dev
+          path: 'packages/vendor-dev/**'
+        - package: vendor
+          path: 'packages/vendor/**'
+
+    steps:
+    # only run if changes are detected in specified path
+    - name: Detect changes
+      uses: dorny/paths-filter@v2.2.0
+      id: changes
+      with:
+        filters: |
+          matched:
+            - ${{ matrix.path }}
+    - name: Set up node
+      if: ${{ steps.changes.outputs.matched == 'true' }}
+      uses: actions/setup-node@v1
+      with:
+        node-version: 12.x
+    - name: Checkout foreman-js
+      if: ${{ steps.changes.outputs.matched == 'true' }}
+      uses: actions/checkout@v2
+    - name: Package tarball
+      if: ${{ steps.changes.outputs.matched == 'true' }}
+      run:  npm pack 
+      working-directory: ${{ github.workspace }}/packages/${{ matrix.package }}
+    - name: Upload tarball locally
```

Comment:
Would be awesome if this action could comment on the PR with the direct download link to the artifact.  (Or at least tell you which packages had changes)

Reply:
this may be prevented because of the issue with secrets and PRs from forks.  :confused:

Reply:
A comment with the download link and some info on how to set it up would be really nice, I can look into it!